### PR TITLE
Cart management

### PR DESCRIPTION
**Pull request by RahullRawat**
- Add to cart
- Remove from cart
- Increase or decrease product quantity
- Total bill
- user can't add products without login

**Login Credentials**
Email - johndoe@gmail.com
Password - johnDoe123

https://deploy-preview-6--aipan.netlify.app/

### CODE REVIEW

File: src/pages/Cart/utils/addToCart.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+
+const addToCart = async (product, cartDispatch, token) => {
+	try {
+		const response = await axios.post(
+			"/api/user/cart",
+			{ product },
+			{ headers: { authorization: token } }
+		);
+		// console.log(response);
```
Comment (AnkurChunekar) [inline src/pages/Cart/utils/addToCart.js]: remove unnecessary comments

File: src/pages/Products/Product.js [hunk 2]
```diff
@@ -59,6 +64,18 @@ export const Product = () => {
 		})();
 	}, []);
 
+	const { cartState, cartDispatch } = useCart();
+	const { token } = useAuth();
+	const navigate = useNavigate();
+
+	const addToCartHandler = async (product) => {
+		if (token) {
+			addToCart(product, cartDispatch, token);
```
Comment (AnkurChunekar) [inline src/pages/Products/Product.js:73]: `addToCart` is already an async function you can remove the async keyword from `addToHandler` function.

**Review by AnkurChunekar — APPROVED** [review]
Everything else looks good, functionalities work as expected, good job @RahullRawat

**Review by mohammed-sas — APPROVED** [review]
1.functionalities are working as stated in description
2. kindly rename the "utils" as "services" because utils folder are used for different purpose
3. folder has been structured well

**Pull request merged**